### PR TITLE
zlib: re-enable "make check"

### PR DESCRIPTION
**Pull request by fanquake**
This was disabled 4 years ago, due to issues with MSAN.
I currently cannot reproduce under MSAN.
If the issue does reproduce, we could better document what it is.

### CODE REVIEW

**Comment by sebpop** [comment]
The change looks good to me.
However I cannot approve as I don't have write access.

**Comment by Alan32Liu** [comment]
> The change looks good to me. However I cannot approve as I don't have write access.

Thanks! I will approve it, then :)

**Review by Alan32Liu — APPROVED** [review]

**Pull request merged**
